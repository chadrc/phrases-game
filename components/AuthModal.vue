<template>
<div class="modal" :class="{'is-active': active}">
  <div class="modal-background"></div>
  <div class="modal-card">
    <header class="modal-card-head">
      <p class="modal-card-title">{{ title }}</p></p>
      <button class="delete" aria-label="close" @click="cancel()"></button>
    </header>
    <section class="modal-card-body">
      <!-- Content ... -->
    </section>
    <footer class="modal-card-foot">
      <button class="button is-success" @click="submit()">{{ submitText }}</button>
      <button class="button" @click="cancel()">{{ cancelText }}</button>
    </footer>
  </div>
</div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      required: true
    },
    submitText: {
      type: String,
      required: true
    },
    cancelText: {
      type: String,
      default: "Cancel"
    },
    active: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    submit() {
      this.$emit("submit");
    },
    cancel() {
      this.$emit("cancel");
    }
  }
};
</script>

<style scoped>
</style>
