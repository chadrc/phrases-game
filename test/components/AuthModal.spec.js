import AuthModal from "@/components/AuthModal.vue";
import { makeVueMock } from "@/test/utils";

describe("SignUpModal", () => {
  test("is vue instance", () => {
    const { wrapper } = makeVueMock(AuthModal, {
      title: "Auth Modal",
      submitText: "Submit"
    });

    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders regular", () => {
    const { wrapper } = makeVueMock(AuthModal, {
      title: "Auth Modal",
      submitText: "Submit"
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("renders active", () => {
    const { wrapper } = makeVueMock(AuthModal, {
      title: "Auth Modal",
      submitText: "Submit",
      active: true
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("custom cancel text", () => {
    const { wrapper } = makeVueMock(AuthModal, {
      title: "Auth Modal",
      submitText: "Submit",
      cancelText: "Don't Submit"
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
