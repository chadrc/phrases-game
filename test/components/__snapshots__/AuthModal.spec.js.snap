// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpModal custom cancel text 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Auth Modal
      </p>
      <p />
       
      <button
        aria-label="close"
        class="delete"
      />
    </header>
     
    <section
      class="modal-card-body"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          Username
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Username"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          Password
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
    </section>
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        Submit
      </button>
       
      <button
        class="button"
      >
        Don't Submit
      </button>
    </footer>
  </div>
</div>
`;

exports[`SignUpModal renders active 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Auth Modal
      </p>
      <p />
       
      <button
        aria-label="close"
        class="delete"
      />
    </header>
     
    <section
      class="modal-card-body"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          Username
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Username"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          Password
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
    </section>
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        Submit
      </button>
       
      <button
        class="button"
      >
        Cancel
      </button>
    </footer>
  </div>
</div>
`;

exports[`SignUpModal renders regular 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Auth Modal
      </p>
      <p />
       
      <button
        aria-label="close"
        class="delete"
      />
    </header>
     
    <section
      class="modal-card-body"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          Username
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Username"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          Password
        </label>
         
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
    </section>
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        Submit
      </button>
       
      <button
        class="button"
      >
        Cancel
      </button>
    </footer>
  </div>
</div>
`;
